docs/getting-started/comparison.md

```diff
@@ -4,11 +4,356 @@ description:
+    case 'decrement':
+      return { count: state.count - action.qty }
+    default:
+      return state
+  }
+}
+
+const countStore = createStore(countReducer)
+```
+
+### Render Optimization
+
+There are no difference between Zustand and Redux. In both you need to do manual
+render optimizations through selectors.
+
+```ts
+import create from 'zustand'
+
+type State = {
+  count: number
+}
+
+type Actions = {
+  increment: (qty: number) => void
+  decrement: (qty: number) => void
+}
+
+const useCountStore = create<State & Actions>((set) => ({
+  count: 0,
+  increment: (qty: number) => set((state) => ({ count: state.count + qty })),
+  decrement: (qty: number) => set((state) => ({ count: state.count - qty })),
+}))
+
+const Component = () => {
+  const count = useCountStore((state) => state.count)
+  const increment = useCountStore((state) => state.increment)
+  const decrement = useCountStore((state) => state.decrement)
+  // ...
+}
+```
+
+```ts
+import { createStore } from 'redux'
+import { useSelector, useDispatch } from 'react-redux'
+
+type State = {
+  count: number
+}
+
+type Action = {
+  type: 'increment' | 'decrement'
+  qty: number
+}
+
+const countReducer = (state: State, action: Action) => {
+  switch (action.type) {
+    case 'increment':
+      return { count: state.count + action.qty }
+    case 'decrement':
+      return { count: state.count - action.qty }
+    default:
+      return state
+  }
+}
+
+const countStore = createStore(countReducer)
+
+const Component = () => {
+  const count = useSelector((state) => state.count)
+  const dispatch = useDispatch()
+  // ...
+}
+```
+
+## Valtio
+
+Zustand and Valtio are state management libraries for React. Here are some
+differences between these two libraries.
+
+### State Model
+
+There is a major difference between Zustand and Valtio. Zustand is based on the
+immutable state model, while Valtio is based on the mutable state model.
+
+```ts
+import create from 'zustand'
+
+const store = create(() => ({ obj: { count: 0 } }))
+
+store.setState((prev) => ({ obj: { count: prev.obj.count + 1 } })
+```
+
+```ts
+import { proxy } from 'valtio'
+
+const state = proxy({ obj: { count: 0 } })
+
+state.obj.count += 1
+```
+
+### Render Optimization
+
+The other difference between Zustand and Valtio is: Valtio makes render
+optimizations through property access. But, with Zustand you need to do manual
+render optimizations through selectors.
+
+```ts
+import create from 'zustand'
+
+const useCountStore = create(() => ({
+  count: 0,
+}))
+
+const Component = () => {
+  const count = useCountStore((state) => state.count)
+  // ...
+}
+```
+
+```ts
+import { proxy, useSnapshot } from 'valtio'
+
+const state = proxy({
+  count: 0,
+})
+
+const Component = () => {
+  const { count } = useSnapshot(state)
+  // ...
+}
+```
+
+## Jotai
+
+Zustand and Jotai are state management libraries for React. Here are some
+differences between these two libraries.
```

Comment:
I feel like by this point we're repeating this line. 

How would you feel about an opening line, right at the top of the page, with something like:

> Zustand is one of many state management libraries for React. On this page we'll discuss Zustand in comparison to some of these libraries, including Redux, Valtio, Jotai, and Recoil.
>
> Each library has its own strengths and weaknesses, and we will compare key differences and similarities between each. 
>